funcx_sdk/funcx/sdk/client.py

```diff
@@ -691,6 +691,19 @@ def register_container(self, location, container_type, name="", description=""):
         r = self.web_client.post("containers", data=payload)
         return r.data["container_id"]
 
+    def build_container(self, container_spec):
+        r = self.web_client.post("containers/build", data=container_spec.to_json())
+        return r.data["container_id"]
+
+    def get_container_build_status(self, container_id):
+        r = self.web_client.get(f"containers/build/{container_id}")
+        if r.http_status == 200:
+            return r["status"]
+        elif r.http_status == 400:
```

Comment:
Should be 404, no

Reply:
I'm currently questioning the choice to unpack and throw away the extra information in the response.  Is there value in pre-parsing the response inside the API over simply returning the response directly?  In particular, if the server ever responds with a non-200 or -404 response, how does the hapless developer debug their situation?

Suggest simply returning the response and let the caller have full control:
```python
def get_container_build_status(self, container_id: UUID | str) -> GlobusHTTPResponse:
    return self.web_client.get(f"containers/build/{container_id}")
```

Reply:
Did fix the error code check. Now looking for 
```python
r.http_status == HTTPStatusCode.NOT_FOUND:
```

Reply:
I added some logging and a better exception message to show the exact error code coming back from the service